src/features/post/components/DisplayPost.jsx

```diff
@@ -0,0 +1,152 @@
+import {
+    Box,
+    Center,
+    Text,
+    Stack,
+    Avatar,
+    useColorModeValue,
+    IconButton,
+    Input,
+    InputGroup,
+    InputRightElement,
+    Button,
+    useToast,
+    HStack,
+    Menu,
+    MenuButton,
+    MenuList,
+    MenuItem,
+    useDisclosure
+} from '@chakra-ui/react';
+import {MdBookmarkBorder, MdFavoriteBorder, MdFavorite} from "react-icons/md";
+import { useDispatch, useSelector } from 'react-redux';
+import { Comment } from './Comment';
+import { likePost, disLikePost, commentOnPost, deletePost } from '../postSlice';
+import { useState } from 'react';
+import { HamburgerIcon } from '@chakra-ui/icons';
+import { CreatePost } from './CreatePost';
+
+export function DisplayPost({post}){
+  const {_id, comments, content, createdAt, likes: {likeCount, likedBy}, username} = post;
+    const {allUsers} = useSelector(store=>store.user);
+    const {user} = useSelector(store=>store.auth);
+    const {firstName, lastName, profilePic} = allUsers.find(user=>user.username===username);
+    const fullName = firstName + " " + lastName;
```

Comment:
Nit pick: Following ES6 you can use template literal here

Reply:
done